### PR TITLE
Improved details page, added relationship between household and owner

### PR DESCRIPTION
**Pull request by Ronandt**
Prepare for lanjiao code

### CODE REVIEW

**Comment by lx0f** [comment]
Lanjia code will be reviewed by @HProsperouS !! Jia you and don't die

image

File: GreenMile/src/Web/Pages/Account/Details.cshtml [hunk 1]
```diff
@@ -7,12 +7,275 @@
 @model Web.Pages.Account.DetailsModel
 @{
 
-    User user = (await userManager.GetUserAsync(HttpContext.User));
+    User user = await userManager.GetUserAsync(HttpContext.User);
 
     Household household = (await householdService.RetrieveHouseholdDetails(user.HouseholdId ?? -1)).Value;
 
 }
+<style>
```
Comment (HProsperouS) [inline GreenMile/src/Web/Pages/Account/Details.cshtml:15]: Maybe is better to put it to an different CSS file so it didn't clog this file

**Review by HProsperouS — COMMENTED** [review]
Add dev.db-wal and dev.db-shm to gitignore

There is a bug when I try to update my last name, without entering the password, it doesn't allow it to be saved. But the flash message said it was saved.

### Steps to recreate the issue:
1. Change your last name to something else in account detail page
2. Click the "Save Changes" button
3. Then the bug will appear

**Comment by Ronandt** [comment]
@HProsperouS Error fixed

File: GreenMile/src/Web/Pages/Register.cshtml.cs [hunk 2]
```diff
@@ -102,20 +105,23 @@ public async Task<IActionResult> OnPostAsync()
...
                 {
                     await _householdService.AddUserToHousehold(userId, (await _householdService.RetrieveHouseholdDetailsByName(HouseholdUiState.JoinHouseholdName)).Value.HouseholdId);
                 }
 
-                _contextAccessor.HttpContext.Session.SetString(SessionVariable.UserName, UserName);
-                _contextAccessor.HttpContext.Session.SetString(SessionVariable.UserId, userId);
-                _contextAccessor.HttpContext.Session.SetString(SessionVariable.HousholdName, user.Household.Name);
+                //_contextAccessor.HttpContext.Session.SetString(SessionVariable.UserName, UserName);
+                //_contextAccessor.HttpContext.Session.SetString(SessionVariable.UserId, userId);
+                //_contextAccessor.HttpContext.Session.SetString(SessionVariable.HousholdName, user.Household.Name);
```
Comment (lx0f) [inline GreenMile/src/Web/Pages/Register.cshtml.cs:124]:
```suggestion
                _contextAccessor.HttpContext.Session.SetString(SessionVariable.UserName, UserName);
                _contextAccessor.HttpContext.Session.SetString(SessionVariable.UserId, userId);
                _contextAccessor.HttpContext.Session.SetString(SessionVariable.HousholdName, user.Household.Name);

```

**Review by lx0f — APPROVED** [review]
Account details update works now 👍
Did the session variables cause any bugs?

**Comment by Ronandt** [comment]
image
fucking please

**Pull request merged**